db/migrate/20191114184656_add_meta_columns_to_users.rb

```diff
@@ -0,0 +1,11 @@
+class AddMetaColumnsToUsers < ActiveRecord::Migration[5.2]
+  def change
+    add_column :users, :last_reaction_at, :datetime, default: "2017-01-01 05:00:00"
```

Comment:
Trying to take a step back and think big picture here. We have A LOT of cached counts on the user model and as is the user table is already pretty big in terms of the number of columns and data stored on it. I wonder if it might be time to break these cached counts out into a related model or handle them in a different way rather than just adding more columns to the user table. 

I also am already a bit concerned about keeping some of the columns we already have updated. The more columns we have that manually need to be touched or updated during updates etc the more likely we are to miss one and then end up with inaccurate data. 

Thoughts?

Reply:
Adding my 2c about the `users` table, I haven't tought yet about the best strategy on keeping counters updated.

I agree with @mstruve in general here, after our conversation around the memory size of users objects in the "comments API caching PR" I started thinking about going around the code and check if and where select too many columns everytime we load users. 

The user table is already quite big:

```sql
PracticalDeveloper_development> select count(*) from information_schema.columns where table_name = 'users';
+---------+
| count   |
|---------|
| 149     |
+---------+
```

I wonder if we shouldn't start moving the counters in proper model(s). 

Most of these counters are not needed when we load users for read purposes, since the default behavior of ActiveRecord `SELECT *` this would mean additional unneeded memory usage everywhere we load users.

If this PR is not urgent I'd like to table it until the next tech meeting, considering the big picture mentioned by Molly

Reply:
Sounds great.

I was actually hoping this is the convo we've had here. I think we could use a bunch more columns like this to build certain types of features off of. I'd love a dedicated table or something. I wasn't sure where we were going to draw the line, but this would free us up a lot. I'm definitely game.